core/arch/arm/plat-rzn1/conf.mk

```diff
@@ -0,0 +1,26 @@
+PLATFORM_FLAVOR ?= rzn1
+
+include core/arch/arm/cpu/cortex-a7.mk
+
+$(call force,CFG_ARM32_core,y)
+$(call force,CFG_TEE_CORE_NB_CORE,2)
+$(call force,CFG_BOOT_SECONDARY_REQUEST,y)
+$(call force,CFG_SECONDARY_INIT_CNTFRQ,y)
+$(call force,CFG_PSCI_ARM32,y)
+$(call force,CFG_16550_UART,y)
+$(call force,CFG_SECURE_TIME_SOURCE_CNTPCT,y)
+$(call force,CFG_WITH_PAGER,n)
+$(call force,CFG_GIC,y)
+$(call force,CFG_SM_PLATFORM_HANDLER,y)
+$(call force,CFG_TA_FLOAT_SUPPORT,n)
+
+ta-targets = ta_arm32
+
+$(call force,CFG_TZDRAM_START,0x88000000)
+$(call force,CFG_TZDRAM_SIZE,0x00A00000)
+$(call force,CFG_SHMEM_START,0x87C00000)
+$(call force,CFG_SHMEM_SIZE,0x00400000)
+$(call force,CFG_TEE_RAM_VA_SIZE,0x00200000)
```

Comment:
These values represent specific memory layout choices on this particular board/RAM combination. It's quite possible for other users of RZ/N1 to want a different layout.

In kernel land, such values would be in DTB rather than in platform driver. Not sure how other OPTEE targets handle this sort of thing.

Reply:
Pretty much hardcoded for all platforms. These values would need to come from a secure DT. Such a beast is partially implemented for a special use case for plat-stm32mp1: a secure DT is embedded in the OP-TEE binary, see `CFG_EMBED_DTB`. I don't think the memory ranges are defined in it however.

So for the time being, these values should probably be defined as `CFG_xxx ?= value` so they can be easily changed at build time.

Reply:
@rfs613

> These values represent specific memory layout choices on this particular board/RAM combination. It's quite possible for other users of RZ/N1 to want a different layout.

Yes, so to handle this scenario I would prefer to add a new `PLATFORM_FLAVOR`. So based on that platform flavour we can define the memory layout.

@jforissier 

> a secure DT is embedded in the OP-TEE binary, see CFG_EMBED_DTB. I don't think the memory ranges are defined in it however.

I think we should be able to retrieve TA memory and shared memory info from a secure DT. But I am not sure how we can make `TEE_LOAD_ADDR` (fixed based address at compile time) configurable via DT. Probably to address this we need to add Position Independent Executable (PIE) feature to OP-TEE OS.

> So for the time being, these values should probably be defined as CFG_xxx ?= value so they can be easily changed at build time.

Sounds reasonable to me, will use it instead.

Reply:
Agreed. We have `-fpie` already, with ASLR.

Reply:
> We have -fpie already, with ASLR.

Okay, I see. So I think it should be doable to have position independent loading of `tee.bin`.